oauth_client.go

```diff
@@ -137,7 +154,7 @@ func (o OAuthClientCreateOptions) valid() error {
 	if !validString(o.HTTPURL) {
 		return errors.New("HTTP URL is required")
 	}
-	if !validString(o.OAuthToken) {
+	if !validString(o.OAuthToken) && *o.ServiceProvider != *ServiceProvider(ServiceProviderBitbucketServer) {
```

Comment:
`o.ServiceProvider` can be nil at the point (it's checked below) -- I think this code could panic?

Reply:
Ah great catch, I will check the service provider first before this check